flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FactoryUtil.java

```diff
@@ -109,11 +109,27 @@ public static DynamicTableSource createTableSource(
 			CatalogTable catalogTable,
 			ReadableConfig configuration,
 			ClassLoader classLoader) {
+		return createTableSource(catalog, objectIdentifier, catalogTable, configuration, classLoader, false);
+	}
+
+	/**
+	 * Creates a {@link DynamicTableSource} from a {@link CatalogTable}.
+	 *
+	 * <p>It considers {@link Catalog#getFactory()} if provided.
+	 */
+	public static DynamicTableSource createTableSource(
```

Comment:
Why don't remove old methods? Looks like they will hides potential bugs.

Reply:
Some tests rely on the old method. But I agree it's better to remove it.